angular-workspace/projects/ni/nimble-angular/src/directives/menu-button/nimble-menu-button.directive.ts

```diff
@@ -46,14 +47,5 @@ export class NimbleMenuButtonDirective {
         this.renderer.setProperty(this.elementRef.nativeElement, 'open', toBooleanProperty(value));
     }
 
-    @Output() public openChange = new EventEmitter<boolean>();
```

Comment:
Note: I removed this `Output` and didn't replace it with a `toggle` `Output` because it causes problems when the Angular `Output` and the native event have the exact same name. The specific issue is that a client app gets their handler called twice -- once triggered by the native element and once triggered by the directive's `Output`. I thought it would be best to simply not have the Angular directive do anything with the event (other than export the type of the event detail).

Reply:
Does the Angular `@Output` gives better typing / inspection for Angular users than them binding to the native events? Maybe if we expect Angular clients to use the `beforetoggle` and `toggle` events then we could stopImmediatePropagation of the native event in the `@HostListener` binding so client bindings with the same name will only be hit once by `@Output` emitted version?

Reply:
The difference in Angular is that `$event` is of type `Event` if nothing is done in the directive and is of type `MenuButtonToggleEventDetail` if we pull the details out in the directive.

If we do nothing in the directive, an Angular user would do something like this in their event handler:
```ts
public onMenuButtonToggle(event: Event) {
    const toggleDetail= (event as CustomEvent).detail as MenuButtonToggleEventDetail;
    // do something with toggleDetail.newState and toggleDetail.oldState
}
```

The typing for an Angular client is better if we do the `as MenuButtonToggleEventDetail` cast in the directive. It just means that they lose access to any other configuration on the original `Event`, but I think that's fine. I'll expose `toggle` and `beforeToggle` as `@Output`s in the directive. Only `toggle` will need to stop propagation since the names are the same for that event/output unlike `before-toggle` and `beforeToggle`. I lean towards not stopping propagation in the `before-toggle` event handler because we don't do that for any of our other handlers within nimble.

Reply:
> I lean towards not stopping propagation in the before-toggle event handler because we don't do that for any of our other handlers within nimble.

Yea I guess that's because none of the other ones have the name conflict so we haven't run into it. It does seem a bit odd that a client could bind to either `beforetoggle` or `beforeToggle` but we can let it run for a while and see if we want to prevent those from leaking through.


> t just means that they lose access to any other configuration on the original Event, but I think that's fine

Can we make an interface that represents the Event to export from components? Then we can keep the normal event data in there. Probably don't need any of the `as` casts either

```ts
interface MenuButtonToggleEvent extends CustomEvent {
   details: MenuButtonToggleEventDetail;
}
```

Reply:
> My personal preference is to leave the event listeners off the directive.

If we don't have an immediate need for them we can kick the can down the road and not expose them yet. If we do need them I think it's worth thinking about (I'd be concerned if clients listen to the event directly which we don't have a great way to type enforce compared to defining `@Output`), but it could be in a follow-up PR to not block action menu work.

Reply:
> Can we make an interface that represents the Event to export from components? Then we can keep the normal event data in there. Probably don't need any of the as casts either

We can make that interface, but it doesn't completely avoid the need for `as` casts because `$event` will be `Event` not `MenuButtonToggleEvent`
```ts
public onMenuButtonToggle(event: Event) {
    const customEvent = event as MenuButtonToggleEvent;
    // do something with customEvent.detail.newState and customEvent.detail.oldState
}
```

>If we don't have an immediate need for them we can kick the can down the road and not expose them yet.

I'm not aware of a need for them. I do think deferring this work is the best option because I think we need to take a holistic approach to deciding which events get turned into Angular Outputs and which we provide event types for. Right now I don't think we're consistent in what we expose in Angular, and I think creating an interface would only increase the inconsistency.

Reply:
Sounds good to me! Maybe a tech debt issue for capturing some of the concerns we want to clean-up would help track it.

Reply:
Created #999